lib/models/addon.js

```diff
@@ -631,14 +631,15 @@ var Addon = CoreObject.extend({
     var templateExtensions = this.registry.extensionsForType('template');
     var addonTreePath = path.join(this.root, this.treePaths['addon']);
     var addonTemplatesTreePath = path.join(this.root, this.treePaths['addon-templates']);
+    var addonTemplatesTreeInAddonTree = addonTreePath.indexOf(addonTemplatesTreePath) === 0;
```

Comment:
If we're going to do this I _think_ we need to `path.normalize` it since you could find yourself traversing out of the context if you were passed `../asdf`.